src/core/ext/lb_policy/pick_first/pick_first.c

```diff
@@ -466,7 +466,9 @@ static grpc_lb_policy *create_pick_first(grpc_exec_ctx *exec_ctx,
     }
 
     memset(&sc_args, 0, sizeof(grpc_subchannel_args));
-    sc_args.server_name = args->server_name;
+    /* The following cast is fine because server_name argument will be copied as
```

Comment:
The one part of this PR that I do like is adding a comment here (and in the corresponding place in round_robin.c) that indicates that the value will be copied as part of subchannel creation.  Even though I'm not in favor of the const changes, these comments seem valuable.